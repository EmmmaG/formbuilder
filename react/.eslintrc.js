module.exports = {
    globals: {},
    env: {
        commonjs: true,
        es6: true,
    },
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    parserOptions: {
        ecmaVersion: 2018,
        ecmaFeatures: {
            globalReturn: false,
        },
        sourceType: 'module',
    },
}
