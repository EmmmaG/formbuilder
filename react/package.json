{
  "name": "hcard-builder",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "^3.0.2",
    "@types/react-dom": "^16.8.3",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-scripts": "2.1.8",
    "typescript": "^3.3.4000"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
  "description": "This project was to conver a fixed format file into csv",
  "main": "index.js",
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@typescript-eslint/eslint-plugin": "^1.5.0",
    "@typescript-eslint/parser": "^1.5.0",
    "@types/enzyme": "^3.1.11",
    "@types/enzyme-adapter-react-16": "^1.0.1",
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-runtime": "^6.25.0",
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^24.1.0",
    "jest-enzyme": "^7.0.1",
    "jest-fetch-mock": "^1.6.5",
    "ts-jest": "^22.4.6"
  },
  "author": "",
  "license": "ISC"
}
