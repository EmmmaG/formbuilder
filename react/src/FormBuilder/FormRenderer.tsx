import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import CircularProgress from '@material-ui/core/CircularProgress'

import { Field } from './FormBuilder'
import { TextInput } from '../Components/TextInputBuilder'
import { RadioSelect } from '../Components/RadioSelectBuilder'
import { Checkbox } from '../Components/CheckboxBuilder'

import TextInputRenderer from '../Components/TextInputRenderer'
import RadioSelectRenderer from '../Components/RadioSelectRenderer'
import CheckboxRenderer from '../Components/CheckboxRenderer'

const styles = (theme: Theme) => ({
    container: {
        minWidth: '400px',
        height: '100vh',
        overflowY: 'auto' as 'auto',
    },
    title: {
        fontSize: '25px',
        margin: '10px',
    },
    loaderContainer: {
		textAlign: 'center' as 'center',
	},
	loader: {
		color: theme.palette.grey[600],
		marginTop: '50px',
	},
})

export interface Props extends WithStyles<typeof styles> {
    fields: Field[]
    onDelete: (id: number) => void
    isLoading?: boolean
}

export class FormRenderer extends React.Component<Props> {
    private onDelete = (id: number) => {
        this.props.onDelete(id)
    }

    private renderTextInput = (field: TextInput) => <TextInputRenderer key={field.id} field={field} onDelete={this.onDelete} />
    private renderRadioSelect = (field: RadioSelect) => <RadioSelectRenderer key={field.id} field={field} onDelete={this.onDelete} />
    private renderCheckbox = (field: Checkbox) => <CheckboxRenderer key={field.id} field={field} onDelete={this.onDelete} />

    private renderForms = () => {
        const { fields } = this.props

        return fields.map(field => {
            if (field.type === 'textInput') {
                return this.renderTextInput(field)
            }
            if (field.type === 'radioSelect') {
                return this.renderRadioSelect(field)
            }
            if (field.type === 'checkbox') {
                return this.renderCheckbox(field)
            }
        })
    }

	public render() {
        const { classes, isLoading } = this.props

		return (
			<Paper className={classes.container}>
                <Typography className={classes.title}>Form Preview</Typography>

                <div>
                    {
                        isLoading && (
                            <div className={classes.loaderContainer}>
                                <CircularProgress
                                    className={classes.loader}
                                    size={100}
                                    variant='indeterminate' />
                            </div>
                        )
                    }
                    {
                        !isLoading && this.renderForms()
                    }
                </div>

            </Paper>
		)
	}
}

export default withStyles(styles)(FormRenderer)
