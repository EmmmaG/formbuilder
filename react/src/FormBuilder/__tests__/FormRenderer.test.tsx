import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import FormRenderer, { Props } from '../FormRenderer'

const fields = [{
    id: 1,
    type: 'textInput',
    question: 'which day you like best?',
    placeholder: 'day',
}, {
    id: 2,
    type: 'checkbox',
    question: 'which day you like?',
    selections: 'Monday###Tuesday###Friday',
}] as any

const props: Props = {
    fields,
    onDelete: jest.fn(),
	classes: idObj,
}

describe('FormRenderer', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<FormRenderer {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
	})
})
