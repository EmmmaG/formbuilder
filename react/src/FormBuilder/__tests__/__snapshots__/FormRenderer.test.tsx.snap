// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRenderer renders correctly 1`] = `
<WithStyles(Paper)
  className="FormRenderer-container-1"
>
  <WithStyles(Typography)
    className="FormRenderer-title-2"
  >
    Form Preview
  </WithStyles(Typography)>
  <div>
    <WithStyles(TextInputRenderer)
      field={
        Object {
          "id": 1,
          "placeholder": "day",
          "question": "which day you like best?",
          "type": "textInput",
        }
      }
      key="1"
      onDelete={[Function]}
    />
    <WithStyles(TextInputRenderer)
      field={
        Object {
          "id": 2,
          "question": "which day you like?",
          "selections": "Monday###Tuesday###Friday",
          "type": "checkbox",
        }
      }
      key="2"
      onDelete={[Function]}
    />
  </div>
</WithStyles(Paper)>
`;
